### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by kaosine** [pulsar-edit/.github#28] [Announcements]
For anybody reading this wanting to help out but unsure where to look, here's all the major repos undergoing change:
 
* [Pulsar Core](https://github.com/pulsar-edit/pulsar) 
  * We have a ton of modernizations they still need to be completed. Checkout our projects tab in the org, or the issues/prs that are current still open.
* [Pulsar Documentation and website](https://github.com/pulsar-edit/pulsar-edit.github.io) 
  * Site is live, and we're trying to get it finished up with the basics. We could use a helping hand in refactoring and rewriting the archived docs as they're copied into their new home. Especially to get it updated with our current progress
* [Atom Backend/API](https://github.com/confused-Techie/atom-community-server-backend-JS) @confused-Techie 's repo, if you have any question for now ask him. More explanation below

# Major Development Items (all these are in some form of progress as of right now but all could use some help here and there)
1. CI reliably functioning across all systems (We're on Cirrus for this!)
    1. We can build for linux and mac, but windows is being impressively stubborn. 
    1. Whether it's bugs or otherwise here, we could use your help!
1. Continue upgrading Electron
   1. We're up to V12 right now thanks to a PR by Mauricio, but we're still behind farther than we should be
   1. Realistically, we should probably stay at least 2-3 versions with in the current release in the future once we are caught up. This would hopefully be the maximum limit
   2. Implement linux auto updates
1. Backend development
   1. Thanks to @confused-Techie we have a api going. But it could use some help getting up to speed.
   1. It needs some features
   1. This is also vital to some workflows like getting the auth token for github with viewing the packages, updating and modifying entries on there.
   1. Improve ppm functionality and keep it on the path
1. Website and documentation
   1. ~This involves bringing the documentation and the site back online~ DONE!
   1. The site needs to be a good landing point outside of the github for all the resources we have.
   1. The documentation needs to be refactored and finished (these have been pulled from the old flightmanual repo)
   1.  Update branding - We have a our name, and possible logo, but we need to make sure everything is consistent.

Check our project boards for further updates!

**Comment by kaosine** [comment]
@pulsar-edit/core @pulsar-edit/documentation I'm bumping this since I know it needs some updates.

**Reply by kaosine** [reply]
Updated a few misc things I'd noticed if anyone wants to see if I missed anything. I'm not sure if @confused-Techie 's api has been brought over in a way that we can see it here. I also remember @autumnblazey was working on cleaning up repo's using branches to do so, not sure where the progress ended on that either. I know the pulsar-edit/flight-manual.atom.io repo needs to be archived however and added to that archival process.

**Reply by meadowsys** [reply]
The archive repo should be done

**Reply by kaosine** [reply]
So maybe you could talk to @confused-Techie or @DeeDeeG and start the process to transfer ownership to the org? I just want to bring this stuff up while I'm thinking of it since I know we had discussed it but never moved on it since the last mention.

**Comment by kaosine** [comment]
I've updated this again, but check our project boards for more up-to-date information!

**Comment by sertonix** [comment]
can this roadmap also include moving to neutal names. Like renaming the global `atom` and methods with atom/pulsar in there name?

**Reply by mauricioszabo** [reply]
This will break 100% of Atom packages, so no...

Like _literally 100%_ - every Atom package needs to use `atom.<something>` to even register...

**Reply by sertonix** [reply]
I mean that the core is renamed but a build-in legacy package would provide the old api. If possible it would be included as dependency if a legacy package is loaded. To do this it would make sense to set env to pulsar for non legacy packages

**Reply by sertonix** [reply]
I still see places where it can't break anything if atom would be replaced in the name. For example in local variables. Can we have a neutral name like `editor` to refactor to?

**Reply by mauricioszabo** [reply]
Editor is fine with me :)

**Reply by sertonix** [reply]
The name `core` suggested by Spiker985 seems better

**Comment by Spiker985** [comment]
I guess if it's local scoping, it doesn't matter much. The global `atom`
declaration should probably be something like `core` instead, when we get
to that point.

That would allow for `core.project.get*` or `core.getReleaseChannel()` and
can't be confused with `TextEditor` instances

**Comment by sertonix** [comment]
Maybe add [teletype](https://github.com/pulsar-edit/teletype) to this list since it is very nice but hasn't been updated/rebranded yet.
A [server](https://github.com/pulsar-edit/teletype-server) exists but the api documentation seems to be missing.
Also it would be nice to make self hosting and custom domains user friendly.
